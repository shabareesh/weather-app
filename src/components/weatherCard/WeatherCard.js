import React from 'react';
import PropTypes from 'prop-types';

import {formatForecastWeatherDate} from "../../utils/utils";

import './weatherCard.scss';

const WeatherCard = ({ data }) => (
  <div className="weather-card">
    <p className="weather-card--date">{formatForecastWeatherDate(data.datetime)}</p>
    <p className="weather-card--temp"><strong>H {data.high_temp}<sup>o</sup></strong><br /><span>L {data.low_temp}<sup>o</sup></span></p>
    <p className="weather-card--description">{data.weather.description}</p>
  </div>
);

WeatherCard.propTypes = {
  data: PropTypes.shape({
    datetime: PropTypes.string.isRequired,
    high_temp: PropTypes.number.isRequired,
    low_temp: PropTypes.number.isRequired,
    weather:PropTypes.shape({
      description:PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default WeatherCard;
