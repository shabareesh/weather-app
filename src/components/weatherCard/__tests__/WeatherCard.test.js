import React from 'react';
import { render } from "@testing-library/react";
import WeatherCard from "../WeatherCard";

describe('WeatherCard', () => {
  test('should generate snapshot', () => {
    const data = {
      datetime: '2020-03-17',
      'high_temp': 13.1,
      'low_temp': 1.1,
      weather: {
        description: 'Partly Clouded',
      },
    };

    const { baseElement } = render(<WeatherCard data={data} />);
    expect(baseElement).toMatchSnapshot();
  });
});
