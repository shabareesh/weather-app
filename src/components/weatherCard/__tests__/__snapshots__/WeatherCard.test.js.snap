// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard should generate snapshot 1`] = `
<body>
  <div>
    <div
      class="weather-card"
    >
      <p
        class="weather-card--date"
      >
        Tue, 3/17
      </p>
      <p
        class="weather-card--temp"
      >
        <strong>
          H 
          13.1
          <sup>
            o
          </sup>
        </strong>
        <br />
        <span>
          L 
          1.1
          <sup>
            o
          </sup>
        </span>
      </p>
      <p
        class="weather-card--description"
      >
        Partly Clouded
      </p>
    </div>
  </div>
</body>
`;
